hibernate-core/src/test/java/org/hibernate/orm/test/annotations/type/dynamicparameterized/MyGenericType.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * Copyright (c) 2010, Red Hat Inc. or third-party contributors as
+ * indicated by the @author tags or express copyright attribution
+ * statements applied by the authors.  All third-party contributions are
+ * distributed under license by Red Hat Inc.
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.hibernate.orm.test.annotations.type.dynamicparameterized;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.usertype.DynamicParameterizedType;
+import org.hibernate.usertype.UserType;
+import org.junit.Assert;
+
+import java.io.Serializable;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Properties;
+
+/**
+ * Test type should only annotate on Entity3.attributes
+ *
+ * @author Yanming Zhou
+ */
+public class MyGenericType implements UserType<Object>, DynamicParameterizedType {
+
+	private ParameterType parameterType;
+
+	@Override
+	public void setParameterValues(Properties params) {
+		parameterType = (ParameterType) params.get(PARAMETER_TYPE);
+        try {
+            Assert.assertEquals( Entity3.class.getDeclaredField("attributes").getGenericType(), parameterType.getReturnedJavaType() );
```

Comment:
I would prefer if you move this assertion statement to a test method into `DynamicParameterizedTypeTest`. You can obtain the type of the attribute through the entity persister.

Reply:
Done.